xmtp_mls/src/groups/group_mutable_metadata.rs

```diff
@@ -110,7 +134,27 @@ impl TryFrom<GroupMutableMetadataProto> for GroupMutableMetadata {
     type Error = GroupMutableMetadataError;
 
     fn try_from(value: GroupMutableMetadataProto) -> Result<Self, Self::Error> {
-        Ok(Self::new(value.attributes.clone()))
+        #[allow(unused_mut)]
+        let mut admin_list: Vec<String>;
+        #[allow(unused_mut)]
+        let mut super_admin_list: Vec<String>;
+        match value.admin_list {
+            Some(inboxes) => {
+                admin_list = inboxes.inbox_ids;
+            }
+            None => return Err(GroupMutableMetadataError::MissingMetadataField),
+        }
+        match value.super_admin_list {
+            Some(inboxes) => {
+                super_admin_list = inboxes.inbox_ids;
+            }
+            None => return Err(GroupMutableMetadataError::MissingMetadataField),
+        }
+        Ok(Self::new(
+            value.attributes.clone(),
+            admin_list,
+            super_admin_list,
+        ))
```

Comment:
Can we do this?
```suggestion
        Ok(Self::new(
            value.attributes.clone(),
            value.admin_list.ok_or(Err(GroupMutableMetadataError::MissingMetadataField))?,
            value. super_admin_list.ok_or(Err(GroupMutableMetadataError::MissingMetadataField)),
        ))
```

Reply:
Thanks @37ng  - that pattern was eluding me for some reason

Reply:
got a more concise version working in a commit here: e08e027ff89e35ba71dee10c8a1e11d25e8f3e07